### PR TITLE
[WIP] Design spike for the namespace api changes

### PR DESCRIPTION
**Pull request by alpe**
See https://github.com/iov-one/weave/issues/231

### CODE REVIEW

File: x/nft/namespace/codec.proto [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+    bytes delegate = 4; // address of user's key or multisig to send ownerFee too. Will be set to owner when empty; optional
+    x.Coin username_product_fee = 5; // required but can be 0
+
+    Fees fees = 7 [(gogoproto.nullable) = false]; // required? can we lift this 1 message level up?
+}
+
+
+message Fees {
+//    x.Coin owner_fee = 1; // only relevant for usernames
```
Comment (Isabello) [inline x/nft/namespace/codec.proto]: ticker and bootstrap too i think

File: x/nft/namespace/codec.proto [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    x.Coin username_product_fee = 2; // costs of a username in this namespace
+}
+
+message IssueTokenMsg {
+    bytes id = 1 [(gogoproto.customname) = "ID"]; // -> key; required
+    bytes owner = 2; // address: user or mulitisig; required
+    repeated nft.ActionApprovals approvals = 3 [(gogoproto.nullable) = false]; // can be empty ?
+
+    bytes delegate = 4; // address of user's key or multisig to send ownerFee too. Will be set to owner when empty; optional
+    x.Coin username_product_fee = 5; // required but can be 0
```
Comment (Isabello) [inline x/nft/namespace/codec.proto]: require 0 or greater

File: x/nft/codec.proto [hunk 3]
```diff
@@ -17,8 +17,11 @@ message ActionApprovals {
 }
 
 message Approval {
-    bytes address = 1;
-    ApprovalOptions options = 2 [(gogoproto.nullable) = false];
+    oneof actor {
+        bytes address = 1;
+        bool all = 2; // everybody when set to true
```
[the remark this answers is not in the document]
Comment (alpe) [reply x/nft/codec.proto]: `owner` is ambivalent too and might not be correct at all as permission might be given to somebody else at the username nft level. what we want instead is something that indicates that not the namespace nft permission model should be used but the username's one. Were we start with the defaults again that could be overwritten.

**Comment by husio** [comment]
I think we can close and delete this pull request now. There will be no discussion on this topic anytime soon.

**Pull request closed without merging**